### ISSUE TITLE
Can't install to Revit 2018.3.1.2

### ISSUE THREAD

**Issue by dmscholten87** [architecture-building-systems/revitpythonshell#82]
Hello,

I've download the RevitPythonShell Package and install it on my computer.
Now there are files on my "AppData\Roaming\RevitPythonShell2018"
- init.py
- startup.py
- RevitPythonShell.xml

And there is a folder on "C:\Program Files (x86)\RevitPythonShell2018" with multiple *dll files.

Now i missing de *addin files and can't find them anywhere. What i'm doing wrong?

I use W10 and Revit 2018.3.1.2.
I had installed 2017.07.24_Setup_RevitPythonShell_2018.msi.

**Comment by CyrilWaechter** [comment]
Hello,

I don't know why you don't have addin files installed but if you want to make addin file yourself manually. A typical addin file for RPS looks like this : 
RevitPythonShell2018.addin
```xml
<?xml version="1.0" encoding="utf-16" standalone="no"?>
<RevitAddIns>
  <AddIn Type="Application">
    <Name>RevitPythonShell</Name>
    <Assembly>C:\Program Files (x86)\RevitPythonShell2018\RevitPythonShell.dll</Assembly>
    <AddInId>3a7a1d24-51ed-462b-949f-1ddcca12008d</AddInId>
    <FullClassName>RevitPythonShell.RevitPythonShellApplication</FullClassName>
  <VendorId>RIPS</VendorId>
  </AddIn>
</RevitAddIns>
```
This addin file is placed in : 
- For a all user install : __%ProgramData%\Autodesk\Revit\Addins\2018__
- For user install : __%AppData%\Autodesk\Revit\Addins\2018__

**Comment by dmscholten87** [comment]
Thx now it will show the button on the ribbon.

But when i add the script with the configure mode and restart Revit i get the next error.

![image](https://user-images.githubusercontent.com/41347566/42866896-3af016ae-8a6e-11e8-883c-c85a4a15f501.png)

I don't know what i doing wrong.

**Comment by dmscholten87** [comment]
I've pyRevit on my computer but the systemmanager blocked some in and outcomming data.
So i can't install RevitMEP direct from the 'extension' button.
Is there an another method to install it to pyRevit if RevitPythonShell doesnt work?

**Comment by CyrilWaechter** [comment]
pyRevit and RevitPythonShell (RPS) are completely independent. As I see it, RPS is a very good way to test analyse Revit, I was using it to deploy my script before pyRevit became so great for this use.

To manually install pyRevit check Manual Installation on pyRevit section on this page : https://eirannejad.github.io/pyRevit/installation/
Video inside from @eirannejad explains well how to proceed with a manual install.

**Comment by dmscholten87** [comment]
I know that it is completely independent but maybe you know a how to :)

I have pyRevit installed but i want the addin RevitMep on pyRevit manual installed.

**Comment by CyrilWaechter** [comment]
For pyRevitMEP process is the same (as explained in pyRevit link from my previous comment) . If you struggle at some point, tell me, but you may [open a separate issue on pyRevitMEP repo](https://github.com/CyrilWaechter/pyRevitMEP/issues) to not pollute RPS bug tracker on subject not related to it.

**Comment by dmscholten87** [comment]
Oke thank you i'm gonna look to the link...

**Comment by VBScab** [comment]
>I don't know why you don't have addin files installed 
For anyone coming here in frustration like I did...

The reason is that the brain-dead installer doesn't copy in the .addin file unless the target folder is present so, for each version, just create "%AppData%\Autodesk\Revit\Addins\201[x]" (where [x] is the last digit of the Revit version) then be sure to run the installation as admi, even if you're logged in with an account which has Administrator-level privileges.

That gets you the .addin file which you can then move/install to "%ProgramData%\Autodesk\Revit\Addins\201[x]" which pretty much every other add-in under the sun uses. 

I need to experiment to see if the folder containing the 2 .py files and the .XML file can be moved to %ProgramData% too but, again, most add-ins are happy there.

**Comment by daren-thomas** [comment]
@VBScab you're right, the installer _is_ brain-dead! Do you have experience writing installers? Would you like to help improve the situation? We could also move to another installer technology (NSIS is one i've started liking recently in another project)...

**Comment by VBScab** [comment]
There's no need to change installer technology, although as an application packager of too many years, I'd recommend avoiding NSIS: corporates don't like it much. Use Windows Installer/MSI. For your purposes, there's no need for sophisticated MSI authoring tools - use the Wix toolset.

All you need to do is add some detection of the installed Revit version (documented on Autodesk's site) and have the installer you have create the target folder.

**Comment by daren-thomas** [comment]
@VBScab would you like to help?

**Comment by VBScab** [comment]
Tricky, as I don't use Wix myself nor do I know which installer technology you currently use.

**Comment by daren-thomas** [comment]
I think we're currently using InnoSetup. I've used WiX before, but if I remember correctly, it has a steep learning curve...

**Comment by VBScab** [comment]
>steep learning curve
Yup. That's why I don't use it! :-) I seem to recall that someone put together a GUI for it, though.

>Inno Setup
I come across quite a few packages which use this. You can write scripts for this, right? In VCL, is it? I can't remember - it's years since I looked at these.

**Comment by eirannejad** [comment]
@daren-thomas @VBScab I'm using AdvancedInstaller for pyRevit and the GUI is really helpful. I have not tested it for automatic builds tho. They give out free licenses for open-source projects.
